common/lib/common-utils/src/timer.ts

```diff
@@ -57,6 +57,31 @@ interface IRunningTimerState extends ITimeout {
     restart?: ITimeout;
 }
 
+/**
+ * The longer the timeout, the less accurate the timeout is exactly going to be
+ * @param timeoutFn - executed when the timeout expires
+ * @param timeoutMs - duration of the timeout in milliseconds
+ * @param setTimerFn - executed to update the timeout if multiple timeouts are required when timeoutMs > maxTimeout
+ * @returns the initial timeout
+ */
+function setLongTimeout(
+    timeoutFn: () => void,
+    timeoutMs: number,
+    setTimerFn: (timer: ReturnType<typeof setTimeout>) => void,
```

Comment:
I think `setTimerFn` should be optional for fire-and-forget type scenarions.

Reply:
nit: should be called  - setTimeoutFn(timeout: ReturnType<typeof setTimeout>) => void;